File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        TreeSet<Integer> ts = new TreeSet<>();
        for(int num:arr){
            ts.add(num);
        }
        HashMap<Integer,Integer> hm = new HashMap<>();
        int rank=1;
        for(int num:ts){
            hm.put(num,rank++);
        }
        for(int i=0;i<arr.length;i++){
            arr[i] = hm.get(arr[i]);
        }
        return arr;
    }
}

File: 1256-rank-transform-of-an-array/1256-rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        dict = {}
        temp = sorted(list(set(arr)))
        for i in range(len(temp)):
            dict[temp[i]] = i+1
        for i in range(len(arr)):
            arr[i] = dict[arr[i]]
        return arr
        